core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -1048,20 +1054,15 @@ class KafkaServer(
    *
    * @return The brokerId.
    */
-  private def getOrGenerateBrokerId(brokerMetadata: RawMetaProperties): Int = {
-    val brokerId = config.brokerId
-
-    if (brokerId >= 0 && brokerMetadata.brokerId.exists(_ != brokerId))
-      throw new InconsistentBrokerIdException(
```

Comment:
@cmccabe There is a test that's meant to catch this exception, but it was buggy: https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/server/ServerGenerateBrokerIdTest.scala#L137

What's the expected behavior now and should the code or test be changed?

Reply:
It just throws a `RuntimException` now, I believe. I don't think it's necessary to have a custom exception for every `meta.properties` issue